extensions/omni_sql/omni_sql.c

```diff
@@ -149,26 +149,58 @@ Datum raw_statements(PG_FUNCTION_ARGS) {
   rsinfo->setResult = tupstore;
 
   char *statement = PG_GETARG_CSTRING(0);
+  bool preserve_transactions = PG_GETARG_BOOL(1);
 
   List *stmts = omni_sql_parse_statement(statement);
 
   ListCell *lc;
+  text *cmd;
+  text *tx = NULL;
   foreach (lc, stmts) {
     switch (nodeTag(lfirst(lc))) {
     case T_RawStmt: {
       RawStmt *raw_stmt = lfirst_node(RawStmt, lc);
-      int line, col;
+
+      int line, col, tx_line, tx_col;
       int actual_start = find_non_whitespace(statement + raw_stmt->stmt_location);
       find_line_col(statement, raw_stmt->stmt_location + actual_start, &line, &col);
-      Datum values[3] = {
-          PointerGetDatum(
-              raw_stmt->stmt_len == 0
-                  ? cstring_to_text(statement + raw_stmt->stmt_location + actual_start)
-                  : cstring_to_text_with_len(statement + raw_stmt->stmt_location + actual_start,
-                                             raw_stmt->stmt_len - actual_start)),
-          Int32GetDatum(line), Int32GetDatum(col)};
-      bool isnull[3] = {false, false, false};
-      tuplestore_putvalues(tupstore, rsinfo->expectedDesc, values, isnull);
+      cmd = raw_stmt->stmt_len == 0
+                ? cstring_to_text(statement + raw_stmt->stmt_location + actual_start)
+                : cstring_to_text_with_len(statement + raw_stmt->stmt_location + actual_start,
+                                           raw_stmt->stmt_len - actual_start);
+
+      if (preserve_transactions) {
+        // begin transaction
```

Comment:
minor: maybe move `//begin` and `//end` inside of the `if/else` blocks to make it clearer to the reader in this case? Looking below it is not clear if we apply the scoping of the comments here consistently and is a bit confusing (at least to me)